target/riscv/translate.c

```diff
@@ -887,6 +934,29 @@ static bool gen_shift(DisasContext *ctx, arg_r *a,
 #ifdef TARGET_CHERI
 /* Must be included first since the helpers are used by trans_rvi.c.inc */
 #include "insn_trans/trans_cheri.c.inc"
+#else
+/* Stubs needed for mode-dependent compressed instructions */
+
+typedef arg_i arg_lc;
+static inline bool trans_lc(DisasContext *ctx, arg_lc *a)
+{
+    g_assert_not_reached();
+    return false;
+}
+
+typedef arg_s arg_sc;
+static inline bool trans_sc(DisasContext *ctx, arg_sc *a)
+{
+    g_assert_not_reached();
+    return false;
+}
+
+typedef arg_i arg_cincoffsetimm;
+static inline bool trans_cincoffsetimm(DisasContext *ctx, arg_cincoffsetimm *a)
+{
+    g_assert_not_reached();
+    return false;
+}
```

Comment:
```suggestion
/* Stubs needed for mode-dependent compressed instructions */
static inline bool trans_stub(DisasContext *ctx, void *a)
{
    g_assert_not_reached();
    return false;
}

#define trans_lc(ctx, a) trans_stub(ctx, a)
#define trans_sc(ctx, a) trans_stub(ctx, a)
#define trans_cincoffsetimm(ctx, a) trans_stub(ctx, a)
```
Not sure if this is better, but it should make it easier to add further stubs?